File: php/examples/payroll/php/coach.php
<?php
include_once('deductions.php');
    class coach extends person{
    use deductions;
        private int $customersAttented;
        private int $bonus;
        private int $totalBonus;

        public function __construct(int $customersAttented, int $bonus, int $totalBonus=0){
            $this->customersAttented=$customersAttented;
            $this->bonus=$bonus;
            $this->totalBonus=intdiv($this->customersAttented,5)*$this->bonus;

            $daysWorked=27;
            $dayValue=45000;
            $salary=$daysWorked*$dayValue;
            $transport= ($salary<2000000)?200000:0;
        
        parent:: __construct("Juanito of Jesus","Perez Iguaran",$daysWorked,$dayValue, $transport,$salary,"Coach"); 
        $this-> initDeductions($salary);
        }
            public function calculateSalary():float{
                 return ($this->getDaysWorked()*$this->getDayValue())-$this->totalDeductions;
            }
            public function getTotalBonus():int{
                return $this->totalBonus;
            }
            public function getCustomersAttented():int{
                return $this->customersAttented;
            }
            public function show():void{
                echo "<h1>Information of the worker: ".$this->getPosition()."</h1>";
                echo "<h2>Position: ".$this->getPosition(). "</h2>";
                echo "<p>Names and last names: " . $this->getNames() ." ". $this->getLastNames() ."</p>";
                echo "<p>Days worked: "  .$this->getDaysWorked(). "</p>";
                echo "<p>Day value: " .$this->getDayValue()."$"."</p>";
                echo "<p>With a salary of: ". $this->getSalary()."$"."</p>";
                echo "<p>With a auxiliary of transport of: ". $this->getTransport()."$". "</p>";
                echo "<p> Con la cantidad de " . $this->getCustomersAttented()." "."personas atendidas"."</p>";
                echo "<p> Gaining a bonus of: " . $this->getTotalBonus(). "$". "</p>";
                $this->showDeductions();
                echo "<p>For obtain a fullpayment of: ". $this->calculateSalary()."$"."</p>";
            }
        }
?>

File: java/payroll/src/main/java/com/maicol/payroll/Coach.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.payroll;

/**
 *
 * @author Maicol
 */
public class Coach extends person {
    private final int customersAttended;
    private final int bonus;
    private final int totalBonus;

    public Coach(int customersAttended, int bonus) {
        super("Juanito of Jesus", "Perez Iguaran", 27, 45000, (27 * 45000 < 2000000) ? 200000 : 0, 27 * 45000, "Coach");
        this.customersAttended = customersAttended;
        this.bonus = bonus;
        this.totalBonus = (customersAttended / 5) * bonus;
    }

    @Override
    public double calculateSalary() {
        return (getDaysWorked() * getDayValue()) - totalBonus;
    }

    public int getTotalBonus() {
        return totalBonus;
    }

    public int getCustomersAttended() {
        return customersAttended;
    }

    @Override
    public void show() {
        System.out.println("Information of the worker: " + getPosition());
        System.out.println("Position: " + getPosition());
        System.out.println("Names and last names: " + getNames() + " " + getLastNames());
        System.out.println("Days worked: " + getDaysWorked());
        System.out.println("Day value: " + getDayValue() + "$");
        System.out.println("With a salary of: " + getSalary() + "$");
        System.out.println("With a transport allowance of: " + getTransport() + "$");
        System.out.println("With " + getCustomersAttended() + " customers attended");
        System.out.println("Gaining a bonus of: " + getTotalBonus() + "$");
        System.out.println("For obtaining a full payment of: " + calculateSalary() + "$");
    }
}
